### DISCUSSION TITLE
wiki.js V3 source code available?

### DISCUSSION THREAD

**Discussion by ryan-carlson** [requarks/wiki#4182] [General]
I understand from the [Blog post](https://blog.js.wiki/news/2021/wiki-js-3-may-2021-update) that V3 is not yet ready for download and testing. 

But is work-in-progress V3 source code available?  

I am interested in the proposed feature for storing assets to S3 as our organization (Fitness Machine Technicians) would like to host video assets in S3 and then serve the content inside of Wiki.js.  The content is confidential so I cannot make access public on S3 (or another platform such as YouTube).  Instead, I want to generate signed S3 URLs that can be used to download the videos.  Ideally, these would be embedded videos using the HTML <video> tag with the <source> referencing a signed S3 URL.

This is why I was curious to look at the source code.  I would like to understand the direction of this feature as I will likely provide a temporary "fork" to implement until an equivalent is available.  And also, I am willing to collaborate on this feature if this help is useful!

**Comment by benjamincarmel** [comment]
The code is here but it's not really usable yet:
https://github.com/Requarks/wiki-vnext

This project only has one main developer who works on it part time so it could be a long while (which is unfortunate because even V2 is still very beta quality and doesn't have any support behind it)

**Reply by ryan-carlson** [reply]
Thanks @nasiralamreeki!

I have been piloting V2 for about a month.  So far, it has everything we have need with the exception of the external media asset feature.  This one is likely a blocker for us because of the cost and performance associated with storing large videos to the DB (works but is not ideal).
